source/plugins/go/input/lib/cadvisor.go

```diff
@@ -1207,3 +1241,30 @@ func getPersistentVolumeMetrics(metricInfo map[string]interface{}, hostName, met
 
 	return metricItems
 }
+
+func ResetWinContainerIdCache() {
+	winContainerIdCache = make(map[string]bool)
```

Comment:
This can cause memory growth until gc kicks in,  instead of creating new map object every time, rather we should delete the entries in the map.

 for key := range winContainerIdCache {
        delete(winContainerIdCache , key)
    }